src/include/duckdb/main/query_profiler.hpp

```diff
@@ -66,15 +67,16 @@ class OperatorProfiler {
 	                      int id);
 	DUCKDB_API OperatorInformation &GetOperatorInfo(const PhysicalOperator &phys_op);
 
-	bool SettingIsEnabled(MetricsType metric) const;
-
 	~OperatorProfiler() {
 	}
 
+	ClientContext &context;
+
 private:
 	//! Whether or not the profiler is enabled
 	bool enabled;
-	profiler_settings_t settings;
+	map<MetricsType, bool> operator_settings;
```

Comment:
Why did we change this to a `map`? Also, if we need this, then, instead of a `map` with `bool`, can we use an unordered set, and if the value is not in the set, then it is false? That should also make `OperatorSettingsCheck` trivial.

Reply:
I like map because it allows me to do this:
```cpp
if (operator_settings[MetricsType::OPERATOR_TIMING]) {
    ...
}
```

But I can also change it to an unordered set.

Reply:
Let's stay with our custom type `typedef unordered_set<MetricsType, MetricsTypeHashFunction> profiler_settings_t;` here. I don't mind only storing a subset of settings in it.
E.g.
```cpp
profiler_settings_t operator_settings;
```